test/WaveEquationCubedSphereTests.jl

```diff
@@ -54,6 +55,14 @@ function compute_pot_to_kin!(w,hnv,qdivu,unv)
    hnv⋅w
 end
 
+"""
+Compute the total mass
+"""
+function compute_mass(L2MM,hh)
+  (L2MM*hh)⋅ones(length(hh))
```

Comment:
This function allocates memory dynamically at two different points:

1) `ones(length(hh))` allocates a whole vector of ones. Note that the dot product against a vector of ones is equivalent to summing up all the entries of the other vector. 
2) `L2MM*hh` creates a temporary output array with the matrix vector product. Use `mul!` and an input workspace vector as we have done in other functions in this script.